stacks/serverless/pixel.prx.org.yml

```diff
@@ -0,0 +1,228 @@
+      Runtime: nodejs8.10
+      Timeout: 30
+      Tags:
+        - Key: Project
+          Value: pixel.prx.org
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: Name
+          Value: !Sub PixelPrxOrg-${EnvironmentType}
+        - Key: "prx:cloudformation:stack-name"
+          Value: !Ref AWS::StackName
+        - Key: "prx:cloudformation:stack-id"
+          Value: !Ref AWS::StackId
+  PixelRestApi:
+    Type: "AWS::ApiGateway::RestApi"
+    Properties:
+      Name: !Sub "${AWS::StackName} API"
+      Description: "API used by PRX Pixel Tracker Lambda"
+      BinaryMediaTypes:
+        - "image/gif"
+  PixelRestApiResource:
+    Type: "AWS::ApiGateway::Resource"
+    Properties:
+      ParentId: !GetAtt PixelRestApi.RootResourceId
+      PathPart: "{proxy+}"
+      RestApiId: !Ref PixelRestApi
+  PixelRestApiRootMethod:
+    Type: "AWS::ApiGateway::Method"
+    Properties:
+      AuthorizationType: NONE
+      HttpMethod: ANY
+      Integration:
+        IntegrationHttpMethod: POST
+        Type: AWS_PROXY
+        Uri: !Sub arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${PixelLambdaFunction.Arn}/invocations
+      RestApiId: !Ref PixelRestApi
+      ResourceId: !GetAtt PixelRestApi.RootResourceId
+  PixelRestApiMethod:
+    Type: "AWS::ApiGateway::Method"
+    Properties:
+      AuthorizationType: NONE
+      HttpMethod: ANY
+      Integration:
+        IntegrationHttpMethod: POST
+        Type: AWS_PROXY
+        Uri: !Sub arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${PixelLambdaFunction.Arn}/invocations
+      RestApiId: !Ref PixelRestApi
+      ResourceId: !Ref PixelRestApiResource
+  PixelRestApiDeployment:
+    Type: "AWS::ApiGateway::Deployment"
+    DependsOn:
+      - PixelRestApiRootMethod
+      - PixelRestApiMethod
+    Properties:
+      RestApiId: !Ref PixelRestApi
+  PixelRestApiStage:
+    Type: "AWS::ApiGateway::Stage"
+    Properties:
+      DeploymentId: !Ref PixelRestApiDeployment
+      Description: Pixel tracker lambda API gateway integration
+      RestApiId: !Ref PixelRestApi
+      StageName: prod
+  PixelRestApiLambdaPermission:
+    Type: "AWS::Lambda::Permission"
+    Properties:
+      Action: "lambda:InvokeFunction"
+      FunctionName: !Ref PixelLambdaFunction
+      Principal: apigateway.amazonaws.com
+      SourceArn: !Sub "arn:aws:execute-api:${AWS::Region}:${AWS::AccountId}:${PixelRestApi}/${PixelRestApiStage}/*"
+  PixelRestApiDomainName:
+    Type: "AWS::ApiGateway::DomainName"
+    Properties:
+      CertificateArn: !Ref Certificate
+      DomainName: !FindInMap [EnvironmentTypeMap, !Ref EnvironmentType, DomainName]
+  PixelRestApiBasePathMapping:
+    Type: "AWS::ApiGateway::BasePathMapping"
+    Properties:
+      DomainName: !Ref PixelRestApiDomainName
+      RestApiId: !Ref PixelRestApi
+      Stage: prod
+  PixelRestApiDomainNameRecordSetGroup:
+    Type: "AWS::Route53::RecordSetGroup"
```

Comment:
Copied this out of the proxy stack ... so I guess we're creating DNS records within the CD stack now?  Or maybe that's just the way you have to do things with the weird API-Gateway cloudfront/dns setup.

Reply:
We have done this in a couple cases. I can take it or leave it, the `prx.tech` hosted zone is not templated, and it would be a massive pain to convert it, so whether we add the entries manually or in these stacks, not sure it matters much. We'll hit some pain if we ever try to deploy multiple instances, though